// @amplitude/analytics-node/lib/esm/node-client.js
import { __assign, __awaiter, __extends, __generator } from "tslib";
import { AmplitudeCore, Destination, returnWrapper, debugWrapper, getClientLogConfig, getClientStates, } from '@amplitude/analytics-core';
import { Context } from './plugins/context';
import { useNodeConfig } from './config';
var AmplitudeNode = /** @class */ (function (_super) {
    __extends(AmplitudeNode, _super);
    function AmplitudeNode() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AmplitudeNode.prototype.init = function (apiKey, options) {
        if (apiKey === void 0) { apiKey = ''; }
        return __awaiter(this, void 0, void 0, function () {
            var nodeOptions;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        // Step 0: Block concurrent initialization
                        if (this.initializing) {
                            return [2 /*return*/];
                        }
                        this.initializing = true;
                        nodeOptions = useNodeConfig(apiKey, __assign({}, options));
                        return [4 /*yield*/, _super.prototype._init.call(this, nodeOptions)];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.add(new Context())];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, this.add(new Destination())];
                    case 3:
                        _a.sent();
                        this.initializing = false;
                        return [4 /*yield*/, this.runQueuedFunctions('dispatchQ')];
                    case 4:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    return AmplitudeNode;
}(AmplitudeCore));
export { AmplitudeNode };
export var createInstance = function () {
    var client = new AmplitudeNode();
    return {
        init: debugWrapper(returnWrapper(client.init.bind(client)), 'init', getClientLogConfig(client), getClientStates(client, ['config'])),
        add: debugWrapper(returnWrapper(client.add.bind(client)), 'add', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.plugins'])),
        remove: debugWrapper(returnWrapper(client.remove.bind(client)), 'remove', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.plugins'])),
        track: debugWrapper(returnWrapper(client.track.bind(client)), 'track', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.queue.length'])),
        logEvent: debugWrapper(returnWrapper(client.logEvent.bind(client)), 'logEvent', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.queue.length'])),
        identify: debugWrapper(returnWrapper(client.identify.bind(client)), 'identify', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.queue.length'])),
        groupIdentify: debugWrapper(returnWrapper(client.groupIdentify.bind(client)), 'groupIdentify', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.queue.length'])),
        setGroup: debugWrapper(returnWrapper(client.setGroup.bind(client)), 'setGroup', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.queue.length'])),
        revenue: debugWrapper(returnWrapper(client.revenue.bind(client)), 'revenue', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.queue.length'])),
        flush: debugWrapper(returnWrapper(client.flush.bind(client)), 'flush', getClientLogConfig(client), getClientStates(client, ['config.apiKey', 'timeline.queue.length'])),
        setOptOut: debugWrapper(client.setOptOut.bind(client), 'setOptOut', getClientLogConfig(client), getClientStates(client, ['config'])),
    };
};
export default createInstance();
